### ISSUE TITLE
v3.0.1 make build error

### ISSUE THREAD

**Issue by UmfintechWtc** [prometheus/prometheus#15550]
### What did you do?

Execute the "make build" command，have some error
![image](https://github.com/user-attachments/assets/0a14ed8f-85e5-42ba-88cd-80a9c6ffd925)
then i start prometheus，access ui，have some error
`[xx@xx prometheus_official]$ ./prometheus --web.listen-address="0.0.0.0:9099" --config.file="/opt/app/mascloud/ansible/prometheus/cmd/prometheus/config.yml"
time=2024-12-07T14:54:33.623+08:00 level=INFO source=main.go:642 msg="No time or size retention was set so using the default time retention" duration=15d
time=2024-12-07T14:54:33.623+08:00 level=INFO source=main.go:689 msg="Starting Prometheus Server" mode=server version="(version=3.0.1, branch=HEAD, revision=1f56e8492c31a558ccea833027db4bd7f8b6d0e9)"
time=2024-12-07T14:54:33.623+08:00 level=INFO source=main.go:694 msg="operational information" build_context="(go=go1.23.0, platform=linux/amd64, user=mas@JumpServer, date=20241207-06:53:34, tags=unknown)" host_details="(Linux 4.19.0-372.26.3.el8.bclinux.x86_64 #1 SMP Mon Dec 5 11:05:59 CST 2022 x86_64 JumpServer (none))" fd_limits="(soft=65535, hard=65535)" vm_limits="(soft=unlimited, hard=unlimited)"
time=2024-12-07T14:54:33.624+08:00 level=INFO source=main.go:770 msg="Leaving GOMAXPROCS=16: CPU quota undefined" component=automaxprocs
time=2024-12-07T14:54:33.627+08:00 level=INFO source=web.go:650 msg="Start listening for connections" component=web address=0.0.0.0:9099
time=2024-12-07T14:54:33.628+08:00 level=INFO source=main.go:1239 msg="Starting TSDB ..."
time=2024-12-07T14:54:33.630+08:00 level=INFO source=tls_config.go:347 msg="Listening on" component=web address=[::]:9099
time=2024-12-07T14:54:33.630+08:00 level=INFO source=tls_config.go:350 msg="TLS is disabled." component=web http2=false address=[::]:9099
time=2024-12-07T14:54:33.633+08:00 level=INFO source=head.go:628 msg="Replaying on-disk memory mappable chunks if any" component=tsdb
time=2024-12-07T14:54:33.633+08:00 level=INFO source=head.go:715 msg="On-disk memory mappable chunks replay completed" component=tsdb duration=19.034µs
time=2024-12-07T14:54:33.633+08:00 level=INFO source=head.go:723 msg="Replaying WAL, this may take a while" component=tsdb
time=2024-12-07T14:54:33.634+08:00 level=INFO source=head.go:795 msg="WAL segment loaded" component=tsdb segment=0 maxSegment=0
time=2024-12-07T14:54:33.634+08:00 level=INFO source=head.go:832 msg="WAL replay completed" component=tsdb checkpoint_replay_duration=42.623µs wal_replay_duration=887.704µs wbl_replay_duration=151ns chunk_snapshot_load_duration=0s mmap_chunk_replay_duration=19.034µs total_replay_duration=1.016347ms
time=2024-12-07T14:54:33.636+08:00 level=INFO source=main.go:1260 msg="filesystem information" fs_type=EXT4_SUPER_MAGIC
time=2024-12-07T14:54:33.636+08:00 level=INFO source=main.go:1263 msg="TSDB started"
time=2024-12-07T14:54:33.637+08:00 level=INFO source=main.go:1446 msg="Loading configuration file" filename=/opt/app/mascloud/ansible/prometheus/cmd/prometheus/config.yml
time=2024-12-07T14:54:33.638+08:00 level=INFO source=main.go:1485 msg="updated GOGC" old=100 new=75
time=2024-12-07T14:54:33.638+08:00 level=INFO source=main.go:1495 msg="Completed loading of configuration file" db_storage=1.811µs remote_storage=2.185µs web_handler=396ns query_engine=1.244µs scrape=791.639µs scrape_sd=33.806µs notify=1.65µs notify_sd=1.214µs rules=1.945µs tracing=9.694µs filename=/opt/app/mascloud/ansible/prometheus/cmd/prometheus/config.yml totalDuration=1.125549ms
time=2024-12-07T14:54:33.638+08:00 level=INFO source=main.go:1224 msg="Server is ready to receive web requests."
time=2024-12-07T14:54:33.638+08:00 level=INFO source=manager.go:168 msg="Starting rule manager..." component="rule manager"`

![image](https://github.com/user-attachments/assets/c9626008-69f8-4685-b38a-9eea864a00d2)

### What did you expect to see?

Successfully built Prometheus, and the Prometheus binary file contains the web UI after construction

### What did you see instead? Under which circumstances?

make build error

### System information

Linux 3.10.0-1160.80.1.el7.x86_64 x86_64

### Prometheus version

```text
[mas@JumpServer prometheus_official]$ ./prometheus --version
prometheus, version 3.0.1 (branch: HEAD, revision: 1f56e8492c31a558ccea833027db4bd7f8b6d0e9)
  build user:       mas@JumpServer
  build date:       20241207-06:53:34
  go version:       go1.23.0
  platform:         linux/amd64
  tags:             unknown
```

### Prometheus configuration file

```yaml
[xx@xx prometheus_official]$ more /opt/app/mascloud/ansible/prometheus/cmd/prometheus/config.yml 
global:
  scrape_interval: 30s
  evaluation_interval: 30s

scrape_configs:
  - job_name: "upn"
    static_configs:
    - targets: ["gitlab:9100"]
```

### Alertmanager configuration file

```yaml
No
```

### Logs

```text
No
```

**Comment by UmfintechWtc** [comment]
## Why did this problem occur? I am not familiar with react. Can someone help me

## go version
`go version go1.23.0 linux/amd64
`
## node version
`v20.18.1
`
## npm version
`10.8.2
`
## make build logs
`
[mas@JumpServer prometheus_official]$ make build
cd web/ui && npm install
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead

added 659 packages in 2m

105 packages are looking for funding
  run `npm fund` for details
npm notice
npm notice New minor version of npm available! 10.8.2 -> 10.9.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.9.2
npm notice To update run: npm install -g npm@10.9.2
npm notice
The old React app has been separated from the npm workspaces setup to avoid
issues with conflicting dependencies. This is a temporary solution until the
 new Mantine-based UI is fully integrated and the old app can be removed.
cd web/ui/react-app && npm install
npm warn deprecated workbox-cacheable-response@6.6.0: workbox-background-sync@6.6.0
npm warn deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm warn deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm warn deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm warn deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm warn deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
npm warn deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm warn deprecated domexception@2.0.1: Use your platform's native DOMException instead
npm warn deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
npm warn deprecated @babel/plugin-proposal-private-methods@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-methods instead.
npm warn deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
npm warn deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
npm warn deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm warn deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm warn deprecated workbox-google-analytics@6.6.0: It is not compatible with newer versions of GA starting with v4, as long as you are using GAv3 it should be ok, but the package is not longer being maintained

added 1843 packages in 2m

285 packages are looking for funding
  run `npm fund` for details
cd web/ui && CI="" npm run build

> prometheus-io@0.300.0 build
> bash build_ui.sh --all

build lezer-promql

> @prometheus-io/lezer-promql@0.300.0 build
> bash ./build.sh

+ npx lezer-generator src/promql.grammar -o src/parser
Wrote src/parser.js and src/parser.terms.js
+ cat src/parser.terms.js
+ bash ./generate-types.sh
+ mkdir -p ./dist
+ indexFile=./dist/index.d.ts
+ [[ -f ./dist/index.d.ts ]]
+ cat
++ sed -E 's/ = [0-9]+/: number/' src/parser.terms.js
+ rollup -c

./src/parser.js → ./dist/index.cjs, ./dist/index.es.js...
created ./dist/index.cjs, ./dist/index.es.js in 65ms
build codemirror-promql

> @prometheus-io/codemirror-promql@0.300.0 build
> npm run build:lib


> @prometheus-io/codemirror-promql@0.300.0 build:lib
> bash ./build.sh

+ tsc --module esnext --target es2018 --outDir dist/esm
+ tsc --module commonjs --target es5 --outDir dist/cjs --downlevelIteration
build react-app

> @prometheus-io/app@0.300.0 build
> react-scripts build

Creating an optimized production build...
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
One of your dependencies, babel-preset-react-app, is importing the
"@babel/plugin-proposal-private-property-in-object" package without
declaring it in its dependencies. This is currently working because
"@babel/plugin-proposal-private-property-in-object" is already in your
node_modules folder for unrelated reasons, but it may break at any time.

babel-preset-react-app is part of the create-react-app project, which
is not maintianed anymore. It is thus unlikely that this bug will
ever be fixed. Add "@babel/plugin-proposal-private-property-in-object" to
your devDependencies to work around this error. This will make this message
go away.
  
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Compiled with warnings.

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/client/prometheus.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/client/prometheus.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/complete/hybrid.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/complete/hybrid.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/complete/index.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/complete/index.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/complete/promql.terms.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/complete/promql.terms.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/index.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/index.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/lint/hybrid.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/lint/hybrid.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/lint/index.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/lint/index.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/index.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/index.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/matcher.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/matcher.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/parser.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/parser.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/path-finder.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/path-finder.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/type.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/type.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/vector.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/parser/vector.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/promql.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/promql.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/function.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/function.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/index.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/index.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/matcher.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/matcher.ts'

Failed to parse source map from '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/vector.ts' file: Error: ENOENT: no such file or directory, open '/opt/app/mascloud/ansible/prometheus_official/web/ui/react-app/node_modules/@prometheus-io/codemirror-promql/src/types/vector.ts'

[eslint] 
src/pages/graph/SeriesName.tsx
  Line 1:21:  'useContext' is defined but never used  @typescript-eslint/no-unused-vars

src/pages/status/Status.tsx
  Line 130:37:  Expected '!==' and instead saw '!='                                                                                                                                                                                                          eqeqeq
  Line 133:6:   React Hook useEffect has a missing dependency: 'setAnimateLogo'. Either include it or remove the dependency array. If 'setAnimateLogo' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

File sizes after gzip:

  497.54 kB  build/static/js/main.46727cf5.js
  53.72 kB   build/static/css/main.a522f891.css

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  https://cra.link/deployment

build mantine-ui

> @prometheus-io/mantine-ui@0.300.0 build
> tsc && vite build

vite v5.4.8 building for production...
✓ 7094 modules transformed.
dist/index.html                       1.83 kB │ gzip:   0.82 kB
dist/assets/codicon-B_nZgZYP.ttf     61.02 kB
dist/assets/index-Bm6642QA.css      223.05 kB │ gzip:  33.11 kB
dist/assets/index-1SrvQFrd.js     2,530.69 kB │ gzip: 798.46 kB

(!) Some chunks are larger than 500 kB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 13.23s
>> bundling npm licenses
rm -f "npm_licenses.tar.bz2" npm_licenses
ln -s . npm_licenses
find npm_licenses/web/ui/node_modules -iname "license*" | tar cfj "npm_licenses.tar.bz2" --files-from=-
rm -f npm_licenses
>> compressing assets
scripts/compress_assets.sh
>> building binaries
/opt/app/mascloud/ansible/gopath/bin/promu build --prefix /opt/app/mascloud/ansible/prometheus_official 
 >   prometheus
 >   promtool

`

**Comment by beorn7** [comment]
Have you followed the [instructions for building from source](https://github.com/prometheus/prometheus?tab=readme-ov-file#building-from-source)?

The version numbers there seem to be outdated.

It is indeed tough to get the React part set up properly. Maybe try our [community channels](https://prometheus.io/community/) to get support.

We use issues to track bugs or feature request, not to provide support. Therefore, I'll close this. However, if you learn how to solve your problems and you have a helpful suggestion how to improve the [instructions for building from source](https://github.com/prometheus/prometheus?tab=readme-ov-file#building-from-source), a PR would be very welcome.
